cli/src/terraform/TerraformTemplateGenerator.py

```diff
@@ -13,8 +13,9 @@ def __init__(self, cluster_model, infrastructure):
 
     def run(self):
         terraform_output_dir = get_terraform_path(self.cluster_model.specification.name)
-        # Remove generated .tf files (not tfstate).
-        delete_files_matching_glob(terraform_output_dir, '*.tf')
+
+        # Only remove epicli generated .tf files, not tfstate or user created files.
+        delete_files_matching_glob(terraform_output_dir, '[0-9][0-9][0-9]_*.tf')
```

Comment:
I am wondering if here we cannot do something smarter to generate our files. Some users could have similar pattern to this.

Reply:
I was thinking about it, but we want to keep this backwards compatible and I highly doubt anyone defined custom Terraform scripts so far. This is more to prepare for: #2982 So we can drop this crappy modules approach with 2 different overlapping Terraform states which is impossible to maintain,.